barretenberg/cpp/src/barretenberg/honk/composer/goblin/full_goblin_composer.test.cpp

```diff
@@ -0,0 +1,152 @@
+  protected:
+    static void SetUpTestSuite()
+    {
+        barretenberg::srs::init_crs_factory("../srs_db/ignition");
+        barretenberg::srs::init_grumpkin_crs_factory("../srs_db/grumpkin");
+    }
+
+    using Curve = curve::BN254;
+    using FF = Curve::ScalarField;
+    using Point = Curve::AffineElement;
+    using CommitmentKey = proof_system::honk::pcs::CommitmentKey<Curve>;
+    using GoblinUltraBuilder = proof_system::GoblinUltraCircuitBuilder;
+    using GoblinUltraComposer = proof_system::honk::GoblinUltraComposer;
+    using ECCVMFlavor = proof_system::honk::flavor::ECCVMGrumpkin;
+    using ECCVMBuilder = proof_system::ECCVMCircuitBuilder<ECCVMFlavor>;
+    using ECCVMComposer = proof_system::honk::ECCVMComposer_<ECCVMFlavor>;
+    using VMOp = proof_system_eccvm::VMOperation<ECCVMFlavor::CycleGroup>;
+
+    /**
+     * @brief Generate a simple test circuit with some ECC op gates and conventional arithmetic gates
+     *
+     * @param builder
+     */
+    void generate_test_circuit(auto& builder)
+    {
+        // Add some arbitrary ecc op gates
+        for (size_t i = 0; i < 3; ++i) {
+            auto point = Point::one() * FF::random_element();
+            auto scalar = FF::random_element();
+            builder.queue_ecc_add_accum(point);
+            builder.queue_ecc_mul_accum(point, scalar);
+        }
+        builder.queue_ecc_eq();
+
+        // Add some conventional gates that utilize public inputs
+        for (size_t i = 0; i < 10; ++i) {
+            FF a = FF::random_element();
+            FF b = FF::random_element();
+            FF c = FF::random_element();
+            FF d = a + b + c;
+            uint32_t a_idx = builder.add_public_variable(a);
+            uint32_t b_idx = builder.add_variable(b);
+            uint32_t c_idx = builder.add_variable(c);
+            uint32_t d_idx = builder.add_variable(d);
+
+            builder.create_big_add_gate({ a_idx, b_idx, c_idx, d_idx, FF(1), FF(1), FF(1), FF(-1), FF(0) });
+        }
+    }
+
+    /**
+     * @brief Mock the interactions of a simple curcuit with the op_queue
+     * @details The transcript aggregation protocol in the Goblin proof system can not yet support an empty "previous
+     * transcript" (see issue #723). This function mocks the interactions with the op queue of a fictional "first"
+     * circuit. This way, when we go to generate a proof over our first "real" circuit, the transcript aggregation
+     * protocol can proceed nominally. The mock data is valid in the sense that it can be processed by all stages of
+     * Goblin as if it came from a genuine circuit.
+     *
+     * @param op_queue
+     */
+    static void perform_op_queue_interactions_for_mock_first_circuit(
+        std::shared_ptr<proof_system::ECCOpQueue>& op_queue)
+    {
+        auto builder = GoblinUltraBuilder(op_queue);
+
+        // Add a mul accum op and an equality op
+        auto point = Point::one() * FF::random_element();
```

Comment:
I wonder it this is faster than sampling a random one.

Reply:
Changed to `Point::random_element()`. Not sure why I did it that way originally